### PR TITLE
Create route to return other user's MPE rooms

### PR DESCRIPTION
**Pull request by Devessier**
Closes https://github.com/AdonisEnProvence/MusicRoom/issues/354

### CODE REVIEW

File: packages/server/app/Controllers/Http/UsersController.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import User from 'App/Models/User';
+import UserService from 'App/Services/UserService';
+import { fromMpeRoomsToMpeRoomSummaries } from '../Ws/MpeRoomsWsController';
+
+interface ComputeUserCanQueryOtherUserMpeRoomsArgs {
+    user: User;
+    queriedUser: User;
+}
+
+async function getIfUserCanQueryOtherUserMpeRooms({
```
Comment (prastoin) [inline packages/server/app/Controllers/Http/UsersController.ts:16]: **praise:** looks great !

File: packages/server/tests/MpeSearchEngine.test.ts [hunk 2]
```diff
@@ -281,6 +285,298 @@ test.group('My MPE Rooms Search', (group) => {
...
+        await Database.beginGlobalTransaction();
+    });
+
+    group.afterEach(async () => {
+        await disconnectEveryRemainingSocketConnection();
+        sinon.restore();
+        await Database.rollbackGlobalTransaction();
+    });
+
+    test("It should send back all other user's public MPE rooms", async (assert) => {
```
Comment (prastoin) [inline packages/server/tests/MpeSearchEngine.test.ts:306]: **Suggestion:** Unless I'm mistaken, the related user playlistVisibility is not covered here ?
In other words the method should be tested here `getIfUserCanQueryOtherUserMpeRooms` ?

File: packages/server/app/Controllers/Http/UsersController.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
```
Comment (prastoin) [inline packages/server/app/Controllers/Http/UsersController.ts:1]: In the last [last merged PR](https://github.com/AdonisEnProvence/MusicRoom/commit/9b8075fcd126181f126cf397e4fc5770f5b58193), we've been creating new controller especially the `UserProfileController` I don't know if we wanna merge both in one ?

Comment (Devessier) [reply packages/server/app/Controllers/Http/UsersController.ts:1]: We can, indeed!
I created this controller before we merged your PR.

File: packages/server/tests/MpeSearchEngine.test.ts (hunk 2 above)
Comment (Devessier) [reply packages/server/tests/MpeSearchEngine.test.ts:306]: You're right, I'll add a test to ensure an error is thrown when querying the rooms of a user that disallows it (either by `PRIVATE` OR `FOLLOWERS_ONLY` prop).

File: packages/server/app/Controllers/Http/UsersController.ts (hunk 3 above)
Comment (prastoin) [reply packages/server/app/Controllers/Http/UsersController.ts:1]: Great !

**Pull request merged**